openadapt/update_utils.py

```diff
@@ -0,0 +1,33 @@
+"""Utility functions for the download app updates."""
+
+import os
+import shutil
+
+
+def set_permissions(path: str) -> None:
+    """Set the permissions of all files to make the executable."""
+    for root, dirs, files in os.walk(path):
+        for dir in dirs:
+            dir_path = os.path.join(root, dir)
+            try:
+                os.chmod(os.path.join(root, dir), 0o755)
+            except PermissionError:
+                print(f"Skipping directory due to PermissionError: {dir_path}")
+            except Exception as e:
+                print(f"An error occurred for directory {dir_path}: {e}")
+        for file in files:
+            file_path = os.path.join(root, file)
+            try:
+                os.chmod(os.path.join(root, file), 0o755)
+            except PermissionError:
+                print(f"Skipping file due to PermissionError: {file_path}")
+            except Exception as e:
+                print(f"An error occurred for file {file_path}: {e}")
+
+
+def unzip_file(file_path: str) -> None:
+    """Unzip a file to the given directory."""
+    if os.path.exists(file_path):
+        shutil.unpack_archive(file_path, os.path.dirname(file_path))
+        set_permissions(os.path.dirname(file_path))
```

Comment:
I do not believe it should be necessary to set permissions in the entire Downloads directory. Is this line even necessary? If so, can we create a new folder and set permissions on that instead?

Reply:
Yess, that would be better. Thanks for the suggestion.

 I have made a few changes. Now the ZIP is extracted to a specified folder. This folder will have the name as OpenAdapt-v{latest_version}. The file will be unzipped inside this folder and the permission of this dir and file inside it will be changed. 

This makes it cleaner too as folder will have specified name with the version.

Something like this.
<img width="578" alt="image" src="https://github.com/OpenAdaptAI/OpenAdapt/assets/48802836/746dee31-d9ac-4b75-a19e-08092b8f0886">